Bug fix 567 voids solo

jonlorax opened the pull request:
Lets get  DTED void work merged, make sarpy great again.

jonlorax committed:
Implement no_voids option in elevation methods

Added 'no_voids' option to methods for handling void values in elevation calculations.

that comment was co-piloted!!
but yes that what i did, c/p !  not fun

jonlorax committed:
Implement tests for no_voids change DTEDReader and DTEDInterpolator

Added tests for DTEDReader and DTEDInterpolator with for no_voids.

jonlorax committed:
Add utility to check DTED files for void data

This utility method checks DTED files for void data, returning results as a dictionary indicating the presence of voids and their indices if requested.

jonlorax committed:
Implement tests for DTED voids checking

Initial commit.
Add tests for checking voids in DTED files by list, directory, and individual file.

jonlorax committed:
Clean up tests for tests of check_for_voids function

clean out debug, python-pep8-itfy

jonlorax committed:
Add handling for predirs in relative path logic for Cross platform

This an github commit for testing on linux,  
working on how to set up testing with files and test data data structure code to handle testing on windows and linux system
so this is functional but dirty

jonlorax committed:
Enhance voids check tests with debug info

Refactor test_check_for_voids_by_list_of_files_true to make it work cross platform as some users will be on Windows and debugging information and update expected results.
Add test for dted with no void data.

jonlorax committed:
Enhance DTED.py with voids warning/logger messages

Added warning/logger messages for DTED data voids.

jonlorax committed:
Changes to allow cross platform testing

Changes to allow cross platfrom testing
Removed my original idea as Jim Greene came up with a more elegant solution.  hey Jim

jonlorax committed:
Enhance test_dted_void.py for cross-platform support

Refactor tests for check_for_voids to improve cross-platform compatibility and remove debug code.
Reorder a test to put more basic on first

jonlorax committed:
Clean up dted_check_voids.py by removing prints

Removed unnecessary print statements and corrected comments.

jonlorax committed:
Fix argument passing in get_elevation method

jonlorax committed:
Correct variable name typo in DTED.py

Fix typo in variable name 'itme' to 'item'.

jonlorax committed:
Enhance geoid.json with comments for DTED paths

Added comments to existing DTED paths for clarity.
Adding more dted files for more thorough testing

jonlorax committed:
Add geoid.json with geoid and DTED file paths

Adding dted files to test new utils dted method,
File is mostly a copy of the geoid.json file under tests/io

jonlorax committed:
Update skip conditions for DTED tests

Update test for existance of dted files in testing

petersontex commented on sarpy/io/DEM/DTED.py:
```diff
@@ -329,9 +329,20 @@ def new_col_int(val, begin):
+        # so:
+        #     ( <row>, <col> )  as has been happening for years
+        #     (( <row>, <col> ), bool)  with the bool for no_voids
+        no_voids = False
```
Discuss implementation

petersontex commented on tests/utils/test_dted_void.py:
```diff
@@ -0,0 +1,95 @@
+import tests
+# Note
+# Linux
+# export SARPY_TEST_PATH='/sar/CuratedData_SomeDomestic/sarpy_test/'
```
Remove this

petersontex commented on tests/utils/test_dted_void.py:
```diff
@@ -0,0 +1,95 @@
+# export SARPY_TEST_PATH='/sar/CuratedData_SomeDomestic/sarpy_test/'
+#
+# windows
+# set SARPY_TEST_PATH=c:\Users\JohnO'Neill\Downloads\  # __DO NOT__ put quotes around it.  you're welcome
```
Remove this too

jonlorax committed:
Clean up comments in test_dted_void.py

Removed unnecessary comments regarding test path setup and execution instructions.

jonlorax committed:
Modify __getitem__ to add no_voids parameter

Updated __getitem__ method to include no_voids parameter for better handling of index adjustments and void values.

jonlorax commented on sarpy/io/DEM/DTED.py:
```diff
@@ -371,17 +382,22 @@ def _repair_values(elevations):
         """
-        out = (elevations & 0x7f_ff).astype(numpy.int16)
-        out *= (-1) ** ((elevations & 0x80_00) != 0)
+        if no_voids and ( elevations & 0xffff) == 0xffff:
```
change parameter name to ignore_voids
and explain bit ops 65353 value

Commit:
renaming no_voids to ignore_voids and clean up math

Commit:
Merge branch 'bug_fix_567_voids_solo' of https://github.com/jonlorax/sarpy into bug_fix_567_voids_solo

cleaning up merge

jonlorax committed:
Remove debugging statements in test_dted.py

Comment out pdb import and set_trace calls.

jonlorax committed:
Enhance print statements in test_dted.py

Updated print statements to include expected input values for clarity.

jonlorax committed:
Clean up comments in dted_check_voids.py

Removed redundant comments about DTEDReader tests.

jonlorax committed:
Refactor test cases for _repair_values method

Updated test cases for _repair_values method to include additional special integer values.

jonlorax committed:
Rename and refactor repair values test

Refactor test_joz_repair_values to test_repair_values and clean up code.
Cleaned up tests,

jonlorax committed:
Change output type to numpy.uint16 for consistancy

keep types same

jonlorax committed:
Improve void handling in DTED data processing

Refactor void handling to check if elevations is a scalar before checking value and logging warnings.

jonlorax committed:
Modify test assertion for void check results

Updated the assertion to check for a subset of dict of results instead of exact match as user DTED directories can be different.
Using a set subset test.

jonlorax committed:
Remove commented debug code and update comments

jonlorax committed:
Merge branch 'staging_v1.3.63' into bug_fix_567_voids_solo

jonlorax committed:
move ignore_voids to dtedreader class and remove ignore_voids parameter to DTED methods, 

Move ignore_voids parameter to be part of DTEDReader and removed ignore_voids parameter from call chain.
sigificant change from web gui ick

jonlorax committed:
Fix tests for moving ignore_voids to class parameter in DTEDReader

Fix tests for moving ignore_voids to class parameter in DTEDReader

jonlorax committed:
Refactor DTED tests to handle ignore_voids move to class variable

Updated DTED tests to handle ignore_voids parameter now in class for elevation calculations and repairs.

jonlorax committed:
Continue changes related to moving ignore_voids to a class member

Continue changes related to moving ignore_voids to a class member, and change for some class methods to handle the ignore_voids

jonlorax committed:
Fix assertion for elevation with ignore_voids

jonlorax committed:
Moved handling of voids in DTED elevation data to init

Refactor DTED class to handle voids in elevation data more efficiently when reading the data into memory

jonlorax committed:
Remove tests for _repair_values

Removed tests for DTEDReader's _repair_values method, as not appropriate now with any void data removed on dted file being read in.

jonlorax committed:
Saving dted demo script

Demo script for dted void work

jonlorax committed:
Refactor DTED class constructor parameters

Removed 'ignore_voids' parameter from constructor and updated docstring.

jonlorax committed:
Revert _repair_values back to a static method

Revert _repair_values method back to a static method.

jonlorax committed:
Add, recover, extend and clean up dted tests

Added tests for DTEDReader and DTEDInterpolator with and without ignoring voids. Updated existing tests for elevation retrieval and fixed expected values.

jonlorax committed:
Update DTEDReader and fix directory path

jonlorax committed:
Delete dted_void_demo.py

jonlorax committed:
Refactor DTEDInterpolator tests for clarity

Updated test cases to use lat_lon_pt for consistency and clarity in testing the DTEDInterpolator's behavior with and without ignoring voids.
